Router & navigation / Solves #5 Solves #6

aliyorulmazdev opened the pull request.

aliyorulmazdev committed:
Initial

aliyorulmazdev committed:
router & navigation initial

aliyorulmazdev committed:
Solves #5 / ProductList is now datatable

aliyorulmazdev committed:
HomePage & ProductDetails Page Styled a bit

dsibinski commented on prExplorer/API/obj/project.assets.json:
```diff
@@ -2021,8 +2021,8 @@
         "net8.0": {
           "targetAlias": "net8.0",
           "projectReferences": {
-            "c:\\Users\\Ermat\\Desktop\\NC Github\\NC\\prExplorer\\Application\\Application.csproj": {
```
Unfortunately, all of those files were added again... you need to merge your `mobx` branch first where these files were deleted, then it should be OK. You will need to rebase this branch before merging this PR - let me know if you have problems with that

dsibinski commented on prExplorer/client-app/src/app/stores/productStore.ts:
```diff
@@ -0,0 +1,125 @@
+  };
+
+  deleteProduct = async (id: string) => {
+    this.loading = true;
```
I've already suggested that in another PR - it would be better if `loading` state is handled with separate actions, not inside `deleteProduct` or `updateProduct`. Then the consumer of this `productStore` can decide whether it wants to start/stop loading. This is not a big deal, but could be useful to have such a habit ;)

dsibinski commented on prExplorer/client-app/src/features/products/dashboard/ProductDashboard.tsx:
```diff
@@ -1,39 +1,27 @@
+  const {loadProducts, productRegistry} = productStore;
+
+  useEffect(() => {
+    if (productRegistry.size <= 1) loadProducts();
```
It would be better if you use `productStore` directly in that case, i.e.:

```suggestion
    if (productRegistry.size <= 1) productStore.loadProducts();
```

currently, when I see this code, I don't know straightaway that `loadProducts()` is from the store, which is quite an important information

dsibinski commented on prExplorer/client-app/src/features/products/dashboard/ProductList.tsx:
```diff
@@ -1,44 +1,139 @@
+          </Table.Row>
+        </Table.Header>
+        <Table.Body>
+          {filteredProducts.map((product) => (
```
That implementation of sorting/filtering does the job. Nicely done!
However, for the future you can take a look at [react-table](https://tanstack.com/table/v8) or [AG grid](https://www.ag-grid.com/react-data-grid/), which provide most of these features out of the box

aliyorulmazdev committed:
Update prExplorer/client-app/src/features/products/dashboard/ProductDashboard.tsx

currently, when I see this code, I don't know straightaway that loadProducts() is from the store, which is quite an important information

Co-authored-by: Dawid Sibiński <dawidsibinski@gmail.com>

dsibinski commented on prExplorer/client-app/src/features/products/dashboard/ProductList.tsx:
```diff
@@ -1,44 +1,139 @@
+                <Image src={product.thumbnail} size="mini" circular style={{ width: '50px', height: '50px', borderRadius: '50%', objectFit: 'cover' }} /> {/* Thumbnail gösterme */}
+              </Table.Cell>
+              <Table.Cell>
+                <Button
```
You can make these "View" and "Delete" buttons a separate components, so you can reuse them on every list/table you will create in the future

dsibinski commented on prExplorer/client-app/src/features/products/details/ProductDetailedChat.tsx:
```diff
@@ -0,0 +1,61 @@
+import React from 'react'
+import {Segment, Header, Comment, Form, Button} from 'semantic-ui-react'
+
+export default observer(function ProductDetailedChat() {
```
hehe, nice feature! ;)

aliyorulmazdev replied:
will try to implement real time chatting via signalR ...

dsibinski replied:
it would be a nice exercise indeed

dsibinski commented on prExplorer/client-app/src/features/products/details/ProductDetailedHeader.tsx:
```diff
@@ -0,0 +1,63 @@
+  color: "white",
+};
+
+interface Props {
```
remember to name your `Props` more specifically, preferably `ProductDetailedHeaderProps` in that case. I already mentioned that in the previous PR

dsibinski reviewed (approved):
@aliyorulmazdev small remarks, it would be nice if you introduce them, but let's not get blocked
